test/support.ts

```diff
@@ -0,0 +1,350 @@
+import fs from 'fs';
+import path from 'path';
+import { inspect, isDeepStrictEqual } from 'util';
+import type { Dialect, Options } from '@sequelize/core';
+import { Sequelize } from '@sequelize/core';
+import { AbstractQueryGenerator } from '@sequelize/core/_non-semver-use-at-your-own-risk_/dialects/abstract/query-generator.js';
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import chaiDatetime from 'chai-datetime';
+import defaults from 'lodash/defaults';
+import isObject from 'lodash/isObject';
+import type { ExclusiveTestFunction, PendingTestFunction, TestFunction } from 'mocha';
+import sinonChai from 'sinon-chai';
+import { Config } from './config/config';
+
+const expect = chai.expect;
+
+const distDir = path.resolve(__dirname, '../lib');
+
+chai.use(chaiDatetime);
+chai.use(chaiAsPromised);
+chai.use(sinonChai);
+
+// Using util.inspect to correctly assert objects with symbols
+// Because expect.deep.equal does not test non iterator keys such as symbols (https://github.com/chaijs/chai/issues/1054)
+chai.Assertion.addMethod('deepEqual', function deepEqual(expected, depth = 5) {
+  // eslint-disable-next-line @typescript-eslint/no-invalid-this -- this is how chai function
+  expect(inspect(this._obj, { depth })).to.deep.equal(inspect(expected, { depth }));
+});
+
+chai.config.includeStack = true;
+chai.should();
+
+// Make sure errors get thrown when testing
+process.on('uncaughtException', e => {
+  console.error('An unhandled exception occurred:');
+  throw e;
+});
+
+let onNextUnhandledRejection: ((error: unknown) => any) | null = null;
+let unhandledRejections: unknown[] | null = null;
+
+process.on('unhandledRejection', e => {
+  if (unhandledRejections) {
+    unhandledRejections.push(e);
+  }
+
+  const onNext = onNextUnhandledRejection;
+  if (onNext) {
+    onNextUnhandledRejection = null;
+    onNext(e);
+  }
+
+  if (onNext || unhandledRejections) {
+    return;
+  }
+
+  console.error('An unhandled rejection occurred:');
+  throw e;
+});
+
+afterEach(() => {
```

Comment:
Why did you remove the check for global.afterEach?

Reply:
All of our tests are run by mocha, so these are always defined